src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1161,6 +1164,106 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
     }
   }
 
+
+  def getWorkspaceStatus(workspaceName: WorkspaceName, userSubjectId: Option[String]): Future[PerRequestMessage] = {
+    asAdmin {
+      dataSource.inFutureTransaction(readLocks = Set(workspaceName)) { txn =>
+        withWorkspaceContext(workspaceName, txn) { workspaceContext =>
+          val STATUS_FOUND = "FOUND"
+          val STATUS_NOT_FOUND = "NOT_FOUND"
+          val STATUS_CAN_WRITE = "USER_CAN_WRITE"
+          val STATUS_CANNOT_WRITE = "USER_CANNOT_WRITE"
+          val STATUS_NA = "NOT_AVAILABLE"
+
+          val bucketName = workspaceContext.workspace.bucketName
+          val rawlsGroupRefs = workspaceContext.workspace.accessLevels
+          val googleGroupRefs = rawlsGroupRefs map { ref =>
+            ref._1 -> containerDAO.authDAO.loadGroup(ref._2, txn)
+          }
+
+          val userRef = userSubjectId match {
+            case Some(id) => containerDAO.authDAO.loadUser(RawlsUserRef(RawlsUserSubjectId(id)), txn)
+            case None => None
+          }
+
+          val userStatus = userRef match {
+            case Some(user) => "FIRECLOUD_USER: " + user.userSubjectId.value -> STATUS_FOUND
+            case None => userSubjectId match {
+              case Some(id) => "FIRECLOUD_USER: " + id -> STATUS_NOT_FOUND
+              case None => "FIRECLOUD_USER: None Supplied" -> STATUS_NA
+            }
+          }
+
+          val rawlsGroupStatuses = rawlsGroupRefs map { ref =>
+            containerDAO.authDAO.loadGroup(ref._2, txn) match {
+              case Some(group) => "WORKSPACE_GROUP: " + group.groupName.value -> STATUS_FOUND
+              case None => "WORKSPACE_GROUP: " + ref._2.groupName.value -> STATUS_NOT_FOUND
+            }
+          }
+
+          val googleGroupStatuses = googleGroupRefs map { ref =>
+            val groupEmail = ref._2.get.groupEmail.value
+            toFutureTry(gcsDAO.getGoogleGroup(groupEmail).map(_.isDefined match {
+              case true => "GOOGLE_GROUP: " + groupEmail -> STATUS_FOUND
```

Comment:
I don't really like the case true/false thing. Either do match { case None/Some } or if/then/else